cpp/include/cudf/detail/aggregation/aggregation.cuh

```diff
@@ -351,6 +340,26 @@ struct update_target_element<Source,
   }
 };
 
+template <typename Source, bool target_has_nulls, bool source_has_nulls>
+struct update_target_element<Source,
+                             aggregation::PRODUCT,
+                             target_has_nulls,
+                             source_has_nulls,
+                             std::enable_if_t<is_product_supported<Source>()>> {
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index,
+                             column_device_view source,
+                             size_type source_index) const noexcept
+  {
+    if (source_has_nulls and source.is_null(source_index)) { return; }
+
+    using Target = target_type_t<Source, aggregation::PRODUCT>;
```

Comment:
Can we put this using declaration in the original template definition of `update_target_element` (with `K` for a generic aggregation)? It seems like the Target type is used in all the specializations anyway.

Reply:
Can't. typedef or using alias doesn't carry forward in template specializations.